extensions/src/main/java/org/pentaho/platform/plugin/action/mondrian/catalog/MondrianCatalogHelper.java

```diff
@@ -280,18 +284,20 @@ protected List<MondrianCatalog> getCatalogs( IPentahoSession pentahoSession ) {
     return SetUniqueList.decorate( catalogs );
   }
 
-  public static int addToCatalog( String baseUrl, boolean enableXmla, String schemaSolutionPath,
+  /**
+   * Add a catalog to the Mondrian Catalog Service.
+   *
+   * @param schemaSolutionPath the solution path to the schema definition, e.g. /etc/mondrian/XXX/schema.xml
+   * @param session            Pentaho session
+   * @param jndiName           the JNDI name of the datasource to use for this catalog, e.g. java:comp/env/jdbc/XXX
+   * @param overwrite          flag to indicate if the catalog should be overwritten if it already exists
+   * @return 0 is success, -1 if an error occurred
+   */
+  public static int addToCatalog( String schemaSolutionPath,
                                   IPentahoSession session, String jndiName, boolean overwrite ) {
-
     IMondrianCatalogService mondrianCatalogService =
       PentahoSystem.get( IMondrianCatalogService.class, "IMondrianCatalogService", session ); //$NON-NLS-1$
```

Comment:
There are still several `PentahoSystem.get` functions scattered through the class. Why not take the chance to refactor them all and make the dependencies on external instances more clear by making them fields initialized in the constructor? The same you did for several others.